libyul/Dialect.h

```diff
@@ -46,8 +46,9 @@ struct BuiltinFunction
 	ControlFlowSideEffects controlFlowSideEffects;
 	/// If true, this is the msize instruction.
 	bool isMSize = false;
-	/// If true, can only accept literals as arguments and they cannot be moved to variables.
-	bool literalArguments = false;
+	/// Same length as the arguments, true for every argument that accepts only
```

Comment:
Does it make sense to do an optimization for the common non-literal case? I.e. using std::optional here, for example?

Reply:
That way we could avoid allocating an array in many cases, at the cost of a slightly more complicated looking check (that the optional has a value)

Reply:
I think it is better to use `std::optional<std::vector<bool>>` than using the implicit assumption that an empty vector means "no literals".

Reply:
That assumption doesn't exist in the current code though. The main assumption is that it's always the same length as the existing arguments.

Independently of that we can certainly use optional though. I am for the idea and will implement it.